l1/l1.go

```diff
@@ -12,39 +12,39 @@ import (
 	"github.com/NethermindEth/juno/l1/contract"
 	"github.com/NethermindEth/juno/service"
 	"github.com/NethermindEth/juno/utils"
-	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/event"
 )
 
 //go:generate mockgen -destination=../mocks/mock_subscriber.go -package=mocks github.com/NethermindEth/juno/l1 Subscriber
 type Subscriber interface {
 	FinalisedHeight(ctx context.Context) (uint64, error)
-	WatchHeader(ctx context.Context, sink chan<- *types.Header) (event.Subscription, error)
 	WatchLogStateUpdate(ctx context.Context, sink chan<- *contract.StarknetLogStateUpdate) (event.Subscription, error)
 	ChainID(ctx context.Context) (*big.Int, error)
 
 	Close()
 }
 
 type Client struct {
-	l1               Subscriber
-	l2Chain          *blockchain.Blockchain
-	log              utils.SimpleLogger
-	network          utils.Network
-	resubscribeDelay time.Duration
-	nonFinalisedLogs map[uint64]*contract.StarknetLogStateUpdate
+	l1                    Subscriber
+	l2Chain               *blockchain.Blockchain
+	log                   utils.SimpleLogger
+	network               utils.Network
+	resubscribeDelay      time.Duration
+	pollFinalisedInterval time.Duration
+	nonFinalisedLogs      map[uint64]*contract.StarknetLogStateUpdate
 }
 
 var _ service.Service = (*Client)(nil)
 
 func NewClient(l1 Subscriber, chain *blockchain.Blockchain, log utils.SimpleLogger) *Client {
 	return &Client{
-		l1:               l1,
-		l2Chain:          chain,
-		log:              log,
-		network:          chain.Network(),
-		resubscribeDelay: 10 * time.Second,
-		nonFinalisedLogs: make(map[uint64]*contract.StarknetLogStateUpdate, 0),
+		l1:                    l1,
+		l2Chain:               chain,
+		log:                   log,
+		network:               chain.Network(),
+		resubscribeDelay:      10 * time.Second,
+		pollFinalisedInterval: 5 * time.Second,
```

Comment:
5 secs is probably too frequent. Block time is 12 seconds on ethereum anyways. Even something like a minute is reasonable IMO.

Reply:
Ah, forgot to approve. It is up to you to tweak the interval or not :+1:

Reply:
Will tweak the interval, thanks.